src/org/exist/xquery/FunctionCall.java

```diff
@@ -362,15 +362,20 @@ public void accept(ExpressionVisitor visitor) {
         visitor.visitFunctionCall(this);
     }
 
-    private class DeferredFunctionCallImpl extends DeferredFunctionCall {
+    private static class DeferredFunctionCallImpl extends DeferredFunctionCall {
```

Comment:
I see, still looks weird to me, see my comment below. I think it best to make this a non-inner class now.